### DISCUSSION TITLE
Only one server from `ensure_installed` is installed

### DISCUSSION THREAD

**Discussion by colepeters** [mason-org/mason.nvim#568] [Q&A] [answered]
I recently got my dotfiles set up on a new machine, but only just now realized that only one of the language servers I have specified via `ensure_installed` is actually installed:
<img width="557" alt="image" src="https://user-images.githubusercontent.com/1713932/196274086-db92bccc-27d8-4a33-aaca-dee69def238f.png">

[Here's my full LSP config for reference](https://github.com/colepeters/dotfiles/blob/main/config/nvim/lua/colepeters/plugins/lsp-config.lua). I'm on NeoVim 0.8 and macOS FWIW.

Is there perhaps a bug with this feature, or have I missed something?

Thanks!

**Comment by williamboman** [comment, marked as answer]
Hello! The `ensure_installed` in `mason-lspconfig` only accepts 1) LSP servers, and 2) server names as defined by [`lspconfig`](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md). You'll have to rename some entries (for example `html-lsp -> html`), and remove others (`prettierd` for example). I added https://github.com/williamboman/mason-lspconfig.nvim/pull/81 which will print a warning notification for invalid entries to make this easier to spot

**Reply by williamboman** [reply]
If you want to automatically install other tools than LSP servers you could try:

1) using https://github.com/WhoIsSethDaniel/mason-tool-installer.nvim
1) add ~5-10 lines of code in your neovim config to manually call [Mason APIs](https://github.com/williamboman/mason.nvim/blob/main/doc/reference.md), this will give you greater flexibility in terms of how and when things are installed

**Reply by colepeters** [reply]
Thanks, this fixes things! I must've installed all the packages manually on my last machine and never checked if my configuration would actually auto install them properly. 😅
